### PR TITLE
Fixes #7876: [API] Create function to "pre-create" a node

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/7876

### CODE REVIEW

File: create-node-api/Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# make all : will build node-external-reports.rpkg.
+# make licensed : will build a license limited version of the plugin
+#
+
+FILES = # add the file that need to be package here like $(NAME)/create-node-api-schema.sql
+SCRIPTS = postinst
+
+include ../makefiles/common-scala-plugin.mk
+
+target/$(NAME)/create-node-api-schema.sql:
```
Comment (VinceMacBuche) [inline create-node-api/Makefile]: I think you can remove this

File: create-node-api/packaging/postinst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+# add post-installation script, like:
+
+# psql -U rudder -h localhost -d rudder -f /opt/rudder/share/plugins/createnodeapi/Createnodeapi-schema.sql
```
Comment (VinceMacBuche) [inline create-node-api/packaging/postinst]: Should we keep this comment ? I think we need the file

File: create-node-api/src/main/scala/com/normation/plugins/createnodeapi/CreateNodeApiPluginDef.scala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  def init = {}
+
+  def oneTimeInit : Unit = {}
+
+  val configFiles = Seq()
+
+  override def apis: Option[LiftApiModuleProvider[_ <: EndpointSchema]] = {
+    if(status.isEnabled()) {
+      println("******** added apu")
```
Comment (VinceMacBuche) [inline create-node-api/src/main/scala/com/normation/plugins/createnodeapi/CreateNodeApiPluginDef.scala]: I think you can remove this ;) and the oh no println below !!

File: create-node-api/src/main/scala/com/normation/plugins/createnodeapi/CreateNodeApi.scala [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        inventory.node.main.id
+      }) match {
+        case Full(id)     => Right(id)
+        case eb: EmptyBox => Left(CreationError.OnSaveInventory((eb?~!"Error during node creation").messageChain))
+      }
+    }
+
+    def accept(template: NodeTemplate): Either[CreationError, NodeSetup] = {
+      val id = template.inventory.node.main.id
+      newNodeManager.accept(id, ModificationId(uuidGen.newUuid), Rest.CreateNodeActor) match {
```
Comment (VinceMacBuche) [inline create-node-api/src/main/scala/com/normation/plugins/createnodeapi/CreateNodeApi.scala]: Don't we want to pass the actor calling the api ? identified through it's api token ?

**Comment by fanf** [comment]
ping @fanf

File: create-node-api/packaging/postinst (hunk 2 above)
Comment (fanf) [reply create-node-api/packaging/postinst]: I prefer to keep it.

File: create-node-api/src/main/scala/com/normation/plugins/createnodeapi/CreateNodeApi.scala (hunk 4 above)
Comment (fanf) [reply create-node-api/src/main/scala/com/normation/plugins/createnodeapi/CreateNodeApi.scala]: yes, certainly so

**Comment by fanf** [comment]
Commit modified

File: create-node-api/Makefile [hunk 5]
```diff
@@ -7,7 +7,4 @@ FILES = # add the file that need to be package here like $(NAME)/create-node-api
 SCRIPTS = postinst
 
 include ../makefiles/common-scala-plugin.mk
-
-target/$(NAME)/create-node-api-schema.sql:
-	cp ./src/main/resources/create-node-api-schema.sql target/$(NAME)/
-
+println
```
Comment (VinceMacBuche) [inline create-node-api/Makefile]: what does println do ? you need at least a target/command so the makefile is valid  ?

Comment (VinceMacBuche) [reply create-node-api/Makefile]: Apart that comment it can be merged !!

**Comment by fanf** [comment]
Commit modified

**Pull request merged**